staging/src/k8s.io/apiextensions-apiserver/pkg/apiserver/schema/defaulting/validation.go

```diff
@@ -47,15 +47,18 @@ func ValidateDefaults(pth *field.Path, s *structuralschema.Structural, isResourc
 		}
 	}
 
-	return validate(pth, s, s, f, false, requirePrunedDefaults)
+	allErr, error, _ := validate(pth, s, s, f, false, requirePrunedDefaults, cel.RuntimeCELCostBudget)
+	return allErr, error
 }
 
 // validate is the recursive step func for the validation. insideMeta is true if s specifies
 // TypeMeta or ObjectMeta. The SurroundingObjectFunc f is used to validate defaults of
 // TypeMeta or ObjectMeta fields.
-func validate(pth *field.Path, s *structuralschema.Structural, rootSchema *structuralschema.Structural, f SurroundingObjectFunc, insideMeta, requirePrunedDefaults bool) (field.ErrorList, error) {
+// If CEL validation cost budget exceeded, the error will be saved in returned arg error instead of allErrs. The caller could handle it separately if needed.
```

Comment:
> If CEL validation cost budget exceeded, the error will be saved in returned arg error instead of allErrs.

Is that accurate? This seems like it is propagating cost-exceeded errors in `allErrs`:
```
if remainingCost < 0 {
	return allErrs, nil, remainingCost
}
```

as an aside, if we *did* propagate a cost error in `err`, that would propagate to callers of `ValidateDefaults`, and would be handled by this block, which has a comment that we never expect to encounter an error:
```go
} else if validationErrors, err := structuraldefaulting.ValidateDefaults(fldPath.Child("openAPIV3Schema"), ss, true, opts.requirePrunedDefaults); err != nil {
	// this should never happen
	allErrs = append(allErrs, field.Invalid(fldPath.Child("openAPIV3Schema"), "", err.Error()))
} else {
```

Reply:
I was left over from previous change. I have removed the incorrect comment. Sorry for that